### ISSUE TITLE
torch.nn.functional.adaptive_max_pool{1|2|3}d can get negative dimension tensor

### ISSUE THREAD

**Issue by cheyennee** [pytorch/pytorch#83071]
### 🐛 Describe the bug

Tensor's shape should be positive integral. However, the second parameter of torch.nn.functional.adaptive_max_pool1d can be set to negative integral which results in the negative integral shape.

```
import torch
results={}
arg_1 = torch.rand([1, 64, 8], dtype=torch.float32)
arg_2 = -255
arg_3 = False
results['res'] = torch.nn.functional.adaptive_max_pool1d(arg_1,arg_2,arg_3,)
print(results['res'].shape)
#torch.Size([1, 64, -255])
```

### Versions

pytorch:1.8.1
python:3.8.3
os:windows11

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by cheyennee** [comment]
I found the same problem on torch.nn.functional.adaptive_max_pool2d.

```
import torch
results={}
arg_1 = torch.rand([1, 64, 8, 9], dtype=torch.float32)
arg_2 = [17,-18]
arg_3 = False
results['res'] = torch.nn.functional.adaptive_max_pool2d(arg_1,arg_2,arg_3,)
print(results['res'].shape)
#torch.Size([1, 64, 17, -18])
```

**Comment by cheyennee** [comment]
I found the same problem on torch.nn.functional.adaptive_max_pool3d.
```
import torch
results={}
arg_1= torch.rand([1, 64, 8, 9, 10], dtype=torch.float32)
arg_2 = [5,-7,9]
arg_3 = False
results['res'] = torch.nn.functional.adaptive_max_pool3d(arg_1,arg_2,arg_3,)
print(results['res'].shape)
#torch.Size([1, 64, 5, -7, 9])
```

**Comment by redmagnos** [comment]
This is fixed at least as of version 1.12.0.
